### DISCUSSION TITLE
SCP and SFTP Autocomplete for hosts and local files

### DISCUSSION THREAD

**Discussion by Willmac16** [blinksh/blink#2013] [Ideas]
It looks like the `scp` and `sftp` commands don’t tab autocomplete local files or hosts.

It would be nice if I’m ever scp-ing something through my IPad to have autocomplete.

**Comment by carloscabanero** [comment]
Not at the moment and it isn’t super obvious to add it because the shell when you start typing doesn’t know if you mean a remote host or a local directory, etc… I’m not sure how other shells help here but please throw any ideas our way.

Until we figure it out, you can enable the Files.app integration for more complex files, etc

**Reply by Willmac16** [reply]
I would suggest listing matching ssh hosts first, followed by local files. Once a colon is in the string then the arg should be assumed to be a remote host, and ideally remote files should be autosuggested.

**Reply by carloscabanero** [reply]
Actually, we may have a better way to do this with the new FileProvider. Will explore.
